### PR TITLE
Load specific file types from Google Drive (issue #4878)

### PR DESCRIPTION
**Pull request by UmerHA**
# Load specific file types from Google Drive (issue #4878)
 Add the possibility to define what file types you want to load from Google Drive.
 
```
 loader = GoogleDriveLoader(
    folder_id="1yucgL9WGgWZdM1TOuKkeghlPizuzMYb5",
    file_types=["document", "pdf"]
    recursive=False
)
```

Fixes ##4878

## Who can review?
Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:
DataLoaders
- @eyurtsev

Twitter: [@UmerHAdil](https://twitter.com/@UmerHAdil) | Discord: RicChilligerDude#7589

### CODE REVIEW

File: langchain/document_loaders/googledrive.py [hunk 1]
```diff
@@ -30,11 +30,12 @@ class GoogleDriveLoader(BaseLoader, BaseModel):
     document_ids: Optional[List[str]] = None
     file_ids: Optional[List[str]] = None
     recursive: bool = False
+    file_types: Optional[str] = None
```
Comment (eyurtsev) [inline langchain/document_loaders/googledrive.py:33]: Hi @UmerHA, for the type annotations, could you replace with an Optional[Sequence[str]] we want to capture the fact that several types are allowed.

```suggestion
    file_types: Optional[Sequence[str]] = None
```

File: langchain/document_loaders/googledrive.py [hunk 2]
```diff
@@ -30,11 +30,12 @@ class GoogleDriveLoader(BaseLoader, BaseModel):
...
 
     @root_validator
     def validate_folder_id_or_document_ids(
         cls, values: Dict[str, Any]
-    ) -> Dict[str, Any]:
+    ) -> Dict[str, Any]:
```
Comment (eyurtsev) [inline langchain/document_loaders/googledrive.py:38]: additional whitespace

File: langchain/document_loaders/googledrive.py [hunk 3]
```diff
@@ -171,13 +195,16 @@ def _load_document_from_id(self, id: str) -> Document:
...
+    def _load_documents_from_folder(self, folder_id: str,
+                                    file_types: Optional[str] = None) -> List[Document]:
         """Load documents from a folder."""
         from googleapiclient.discovery import build
 
         creds = self._load_credentials()
         service = build("drive", "v3", credentials=creds)
         files = self._fetch_files_recursive(service, folder_id)
+        if file_types:
+            files = [file for file in files if file["mimeType"] in file_types]
```
Comment (eyurtsev) [inline langchain/document_loaders/googledrive.py:207]: is `mimeType`  a mime type (application/pdf) or a simplified file type (e.g,. `pdf`) ?

Comment (UmerHA) [reply langchain/document_loaders/googledrive.py]: `mimeType` is a full mime type (application/pdf). If the user gave simplified file types (eg `pdf`), they were translated to full mime types in `validate_inputs`

File: langchain/document_loaders/googledrive.py (hunk 1 above)
Comment (UmerHA) [reply langchain/document_loaders/googledrive.py]: Done ✅

**Comment by eyurtsev** [comment]
merging here: https://github.com/hwchase17/langchain/pull/4926

**Pull request closed without merging**